### PR TITLE
feat: Add software team tool lib for DI

### PR DESCRIPTION
**Pull request by iorisa**
**Features**
Add software team tool lib for DI

### CODE REVIEW

File: metagpt/tools/libs/software_development.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+from metagpt.utils.common import any_to_str
+
+
+@register_tool(tags=["software company", "ProductManager"])
+async def write_prd(idea: str, project_path: Optional[str | Path] = None) -> Path:
+    """Writes a PRD based on user requirements.
+
+    Args:
+        idea (str): The idea or concept for the PRD.
+        project_path (Optional[str|Path], optional): The path to an existing project directory created by `write_prd`.
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:20]: The path to an existing project directory created by `write_prd`.
->
The path to an existing project directory used by `write_prd`.

File: metagpt/tools/libs/software_development.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+from pathlib import Path
+from typing import Optional
+
+from metagpt.const import BUGFIX_FILENAME, REQUIREMENT_FILENAME
+from metagpt.schema import BugFixContext, Message
+from metagpt.tools.tool_registry import register_tool
+from metagpt.utils.common import any_to_str
+
+
+@register_tool(tags=["software company", "ProductManager"])
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:14]: The recommended description of tag is software development, which corresponds to the file name.

File: metagpt/tools/libs/software_development.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        ctx.config.project_path = Path(project_path)
+        ctx.config.inc = True
+    role = ProductManager(context=ctx)
+    msg = await role.run(with_message=Message(content=idea, cause_by=UserRequirement))
+    await role.run(with_message=msg)
+    return ctx.repo.workdir
+
+
+@register_tool(tags=["software company", "Architect"])
+async def write_design(project_path: str | Path) -> Path:
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:55]: A clear upstream file name is required, and a clear return file name is required. The current input and output granularity is too large

File: metagpt/tools/libs/software_development.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    from metagpt.roles import ProductManager
+
+    ctx = Context()
+    if project_path:
+        ctx.config.project_path = Path(project_path)
+        ctx.config.inc = True
+    role = ProductManager(context=ctx)
+    msg = await role.run(with_message=Message(content=idea, cause_by=UserRequirement))
+    await role.run(with_message=msg)
+    return ctx.repo.workdir
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:51]: The granularity of this return is too large. Logically it should return a file name.

File: metagpt/tools/libs/software_development.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    ctx = Context()
+    ctx.set_repo_dir(project_path)
+
+    role = Architect(context=ctx)
+    await role.run(with_message=Message(content="", cause_by=WritePRD))
+    return project_path
+
+
+@register_tool(tags=["software company", "Architect"])
+async def write_project_plan(project_path: str | Path) -> Path:
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:85]: A clear upstream file name is required, and a clear return file name is required. The current input and output granularity is too large

File: metagpt/tools/libs/software_development.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+    ctx = Context()
+    ctx.set_repo_dir(project_path)
+
+    role = ProjectManager(context=ctx)
+    await role.run(with_message=Message(content="", cause_by=WriteDesign))
+    return project_path
+
+
+@register_tool(tags=["software company", "Engineer"])
+async def write_codes(project_path: str | Path, inc: bool = False) -> Path:
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:115]: A clear upstream file name is required, and a clear return file name is required. The current input and output granularity is too large

File: metagpt/tools/libs/software_development.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    role = Engineer(context=ctx)
+    msg = Message(content="", cause_by=WriteTasks, send_to=role)
+    me = {any_to_str(role), role.name}
+    while me.intersection(msg.send_to):
+        msg = await role.run(with_message=msg)
+    return project_path
+
+
+@register_tool(tags=["software company", "QaEngineer"])
+async def run_qa_test(project_path: str | Path) -> Path:
```
Comment (geekan) [inline metagpt/tools/libs/software_development.py:157]: A clear upstream file name is required, and a clear return file name is required. The current input and output granularity is too large

File: metagpt/tools/libs/software_development.py (hunk 1 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py:20]: fixed.

File: metagpt/tools/libs/software_development.py (hunk 2 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py:14]: fixed

File: metagpt/tools/libs/software_development.py (hunk 3 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py]: `project_path` is renamed to `prd_path`

File: metagpt/tools/libs/software_development.py (hunk 4 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py]: fixed

File: metagpt/tools/libs/software_development.py (hunk 5 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py]: fixed. return task path

File: metagpt/tools/libs/software_development.py (hunk 6 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py]: fixed. input task_path and return src path

File: metagpt/tools/libs/software_development.py (hunk 7 above)
Comment (iorisa) [reply metagpt/tools/libs/software_development.py]: fixed, input src path and return test path

**Pull request closed without merging**